lib/internal/Magento/Framework/GraphQl/Query/Resolver/Argument/FieldEntityAttributes.php

```diff
@@ -25,28 +25,37 @@ class ProductEntityAttributesForAst implements FieldEntityAttributesInterface
     /**
      * @var array
      */
-    private $additionalAttributes = ['min_price', 'max_price', 'category_id'];
+    private $additionalAttributes;
+
+    /**
+     * @var string
+     */
+    private $configElementName;
 
     /**
      * @param ConfigInterface $config
+     * @param string $configElementName
      * @param array $additionalAttributes
      */
     public function __construct(
         ConfigInterface $config,
+        string $configElementName,
         array $additionalAttributes = []
-    ) {
+    )
+    {
         $this->config = $config;
-        $this->additionalAttributes = array_merge($this->additionalAttributes, $additionalAttributes);
+        $this->configElementName = $configElementName;
+        $this->additionalAttributes = $additionalAttributes;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getEntityAttributes() : array
+    public function getEntityAttributes(): array
     {
-        $productTypeSchema = $this->config->getConfigElement('SimpleProduct');
+        $productTypeSchema = $this->config->getConfigElement($this->configElementName);
```

Comment:
Please eliminate all `product` occurrences in this file since now it is generic.